src/model/utils.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const DB = require('./db');
+let Game = null;
+let Board = null;
+
+const init = Promise.resolve()
+	.then(() => {
+		// This works around the incompletely constructed Game/Board objects issue
+		Game = require('./Game');
+		Board = require('./Board');
+	});
+
+const deserializeBoardOrGame = (data) => {
+	if (data.GameID) {
+		return new Game(data);
+	}
+	delete data.gameDescription;
+	return new Board(data);
+};
+
+module.exports = {
+	getBoardOrGame: (id) => init.then(() => {
+		return DB.knex('Boards')
```

Comment:
Should this belong here? I'm uncertain.

Reply:
@Bluenaxela  your thoughts?

Reply:
I put it here because both Board and game want the functionality, Board is parent of Game, so if i try to import Game in Board to put the functionality in the base class i get an error because circular class dependency.

i tried a few different fixes to it but the only one that both worked and didn't squick me out was to put the imports in a promise so they would resolve after the class construction happened. hence the cryptic comment in the creation of the `init` promise....

I'm open to other approaches, this works but it seems pretty hackish to me.

Reply:
I thought more about this and figured out a better way. new commit incoming.